src/graph/network/communicator.h

```diff
@@ -6,112 +6,144 @@
 #ifndef DGL_GRAPH_NETWORK_COMMUNICATOR_H_
 #define DGL_GRAPH_NETWORK_COMMUNICATOR_H_
 
+#include <dmlc/logging.h>
+
 #include <string>
 
+#include "msg_queue.h"
+
 namespace dgl {
 namespace network {
 
 /*!
  * \brief Network Sender for DGL distributed training.
  *
- * Sender is an abstract class that defines a set of APIs for sending 
- * binary data over network. It can be implemented by different underlying 
- * networking libraries such TCP socket and ZMQ. One Sender can connect to 
- * multiple receivers, and it can send data to specified receiver via receiver's ID.
+ * Sender is an abstract class that defines a set of APIs for sending binary 
+ * data message over network. It can be implemented by different underlying 
+ * networking libraries such TCP socket and MPI. One Sender can connect to 
+ * multiple receivers and it can send data to specified receiver via receiver's ID.
  */
 class Sender {
  public:
+  /*!
+   * \brief Sender constructor
+   * \param queue_size size (bytes) of message queue. 
+   * Note that, the queue_size parameter is optional.
+   */
+  explicit Sender(int64_t queue_size = 0) {
+    CHECK_GE(queue_size, 0);
+    queue_size_ = queue_size;
+  }
+
   virtual ~Sender() {}
 
   /*!
-   * \brief Add receiver address and it's ID to the namebook
-   * \param ip receviver's IP address
-   * \param port receiver's port
+   * \brief Add receiver's address and ID to the sender's namebook
+   * \param addr Networking address, e.g., 'socket://127.0.0.1:50091', 'mpi://0'
    * \param id receiver's ID
    */
-  virtual void AddReceiver(const char* ip, int port, int id) = 0;
+  virtual void AddReceiver(const char* addr, int id) = 0;
```

Comment:
Please update the docstring about whether this API is multi-thread safe or not.

Reply:
same